### DISCUSSION TITLE
Shutter - AutoSetup (Only Shelly plus 2PM, ESP32 based)

### DISCUSSION THREAD

**Discussion by Mark-Muc** [arendst/Tasmota#18548] [Q&A]
Hello . just built a version:
```
Tasmota Version	12.5.0(TasmoCompiler-esp32generic)
Erstellungs-Datum & -Uhrzeit	2023.05.01 06:35:22
Core-/SDK-Version	2_0_7/4.4.4.20230310
```
of course using;
```
Shutter / Blind selected and
#define USE_SHUTTER_ESP32
#define MOTOR_STOP_TIME 750
```
the command `shuttersetup` exists.
I deleteed also the file: .`drvset027`
drive the shutter down like descibed: 

```
AutoSetup (Only Shelly plus 2PM, ESP32 based)[~](https://tasmota.github.io/docs/Blinds-and-Shutters/#autosetup-only-shelly-plus-2pm-esp32-based)
The shelly plus has enough memory and a power measuring unit to setup the shutter in convinent way. First you must callibrate your mechanical endstops of the shutter. Please do as descibed in the documentation of your shutter motors to ensure the shutter will stop at the endpoint correctly.

Then close the shutter until endstop is reached (repeat: backlog shuttersetopen;shutterclose until closed) - interlock 1,2 - interlock on - shutterrelay1 1 - shuttersetup (shutter will start moving....)

After setup is started the shutter will move to the upper endpoint and close again. If the shutter stops somewhere in the middle, try again. After setup you can use your shutter. Initial callibration for ShutterSetHalfway is 70. If you want to have it more accurate: backlog shutterclose;ShutterSetHalfway 50 and follow instructions below.
```
The problem is . if i enter the command: `shuttersetup `
The shutter moves up for a second and set all close/upen duration to 1 sec
Tried it several times ...
My currents are:
```
close current 0,588A
open current 0,571A
```
It seems to me, that the time Tasmota wait for currentstabilistion is too short!!
Someone have an idea whats goning wrong?

Mark
BTW; NICE feature !!.. I think it olaso can work on a **_Sonoff dual R3_**.

**Comment by steiny2k** [comment]
Hi All,

nice feature indeed. I was wondering what is preventing it to be enabled for an ESP8266 based Shelly 2.5. @stefanbode, do you see any chance this could be back-ported?

Thanks a lot and have a great day,
Seb

**Reply by stefanbode** [reply]
Theoretically for sure. But Theo will kill me if I add 2k of code just for a one time setup routine of the shutter. The ESP8266 is limited in space and we have implemented quite a lot of stuff. What could be possible is a compiler option that you build your own ESP8266 version. I do not know if this is convenient either. All enhancements to shutters will be ESP32 only from now on. ESP8266 will be legacy and only get bug fixes. For example the shutterbuttons on ESP32 are insane better.

**Comment by stefanbode** [comment]
thanks for bringing this up with the low current. Yes, stabilization could be the reason. The setup is still experimental to get feedback from the field like yours. I will let you know when there is a new version worth to try again. Regarding the "Sonoff Dual R3": It might be that this already works. I only check in general if there is a current senor available delivering data. There is no hard binding to the shelly.

**Reply by Mark-Muc** [reply]
Sonoff Dual R3 do have a current mesurment, so it should work:
![image](https://user-images.githubusercontent.com/67410847/236602716-f5174310-07be-49a2-95c8-b71de68cb516.png)
![image](https://user-images.githubusercontent.com/67410847/236602719-e0fc44e7-9df2-48bf-9332-19e3fe0be427.png)

**Comment by stefanbode** [comment]
> to me, that the time Tasmota wait for currentstabilis

I have seen this issue if Power1 is not connected to relay1. when going up with relay1 the O1 power should show a value. But let me do a fix on the code because this seems to be a common problem. I had the same but can debug :-)

**Comment by stefanbode** [comment]
Submitted a patch to overcome the issue with the setup https://github.com/arendst/Tasmota/pull/18586.

It should now run smoothly. Please be aware you must start the open and close through button or better command

**Reply by Mark-Muc** [reply]
Today i got so ne shelly  plus 2PM and i tested the new SHUTTER_ESP32 on 13.0.1 dev ... It works fine on shelly's .. no problem also set up with `shuttersetup `.. worked like charm... THX to it and it safes a lot of time!! 
then I tried on 2 Sonoff's Dual R3 v2 .. it take me longer for setting up .. after 3 times i got the window shutter working with `shuttersetup` 
on the door shutter I can't get it to work .. the `shutteropenduration1 `was mostly set correct but then the command for closing was missing and if i did it manually (close) the message in the consola apear the but `shuttercloseduration1 ` was set to 1 sec ...
I'd say it works in shelly's perfect ..on 
**Sonoff Dual R3 v2 it won't work!!**

Again also thx for adding the value `shuttermotorstop` .. so i don't have to set `MOTOR_STOP_TIME`  during compiling.
Regards and thankx u.
Mark_Muc

**Reply by stefanbode** [reply]
Can you share some logs from the dual. Maybe I can find the reason for the different behavior.

**Comment by Gifford47** [comment]
@stefanbode Will the shuttersetup be available for the shelly 2.5 as well? currently i get "command unknown" on shelly 2.5 with tasmota 13.0.0

**Reply by stefanbode** [reply]
This is dedicated to the Esp32 series. This does mean it must be a Shelly PLUS. we do not have enough space for a setup only procedure.

**Comment by Mark-Muc** [comment]
sure.. what u like to have? first of all i will update the module in question
________________________________

**Reply by stefanbode** [reply]
Best is with loglevel 4 the calibration process until it stops the automation. You said you had to give manually the command to close. This I do not need. The error is before.

**Reply by Mark-Muc** [reply]
Hello stefanbode!!
I did, as I said. I updated the module with the same version and made a logging like u asked for.
Now it worked fine, without any problems:
Result:
```
08:30:53.071 CMD: ShutterCloseDuration1
08:30:53.080 MQT: stat/Rollo-1-OG-Schlaf-Tuer/RESULT = {"ShutterCloseDuration1":29.0}

08:31:15.688 CMD: ShutterOpenDuration1
08:31:15.696 MQT: stat/Rollo-1-OG-Schlaf-Tuer/RESULT = {"ShutterOpenDuration1":30.4}

```
Loging:
[loging.txt](https://github.com/arendst/Tasmota/files/11975404/loging.txt)

Thanks for ur help & Regards
Mark_Muc
